src/server/v2.0/handler/artifact.go

```diff
@@ -103,11 +85,98 @@ func (a *artifactAPI) ListArtifacts(ctx context.Context, params operation.ListAr
 }
 
 func (a *artifactAPI) GetArtifact(ctx context.Context, params operation.GetArtifactParams) middleware.Responder {
-	// TODO implement
-	return operation.NewGetArtifactOK()
+	// set option
+	option := option(params.WithTag, params.WithImmutableStatus,
+		params.WithLabel, params.WithScanOverview, params.WithSignature)
+
+	// get the artifact
+	artifact, err := a.artCtl.GetByReference(ctx, fmt.Sprintf("%s/%s", params.ProjectName, params.RepositoryName), params.Reference, option)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+	return operation.NewGetArtifactOK().WithPayload(artifact.ToSwagger())
 }
 
 func (a *artifactAPI) DeleteArtifact(ctx context.Context, params operation.DeleteArtifactParams) middleware.Responder {
-	// TODO implement
+	artifact, err := a.artCtl.GetByReference(ctx, fmt.Sprintf("%s/%s", params.ProjectName, params.RepositoryName), params.Reference, nil)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+	if err = a.artCtl.Delete(ctx, artifact.ID); err != nil {
+		return a.SendError(ctx, err)
+	}
 	return operation.NewDeleteArtifactOK()
 }
+
+func (a *artifactAPI) CreateTag(ctx context.Context, params operation.CreateTagParams) middleware.Responder {
+	art, err := a.artCtl.GetByReference(ctx, fmt.Sprintf("%s/%s", params.ProjectName, params.RepositoryName),
+		params.Reference, &artifact.Option{
+			WithTag: true,
+		})
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+	tag := &artifact.Tag{}
+	tag.RepositoryID = art.RepositoryID
+	tag.ArtifactID = art.ID
+	tag.Name = params.Tag.Name
+	tag.PushTime = time.Now()
+	if _, err = a.artCtl.CreateTag(ctx, tag); err != nil {
+		return a.SendError(ctx, err)
+	}
+	// TODO set location header?
+	return operation.NewCreateTagCreated()
+}
+
+func (a *artifactAPI) DeleteTag(ctx context.Context, params operation.DeleteTagParams) middleware.Responder {
+	artifact, err := a.artCtl.GetByReference(ctx, fmt.Sprintf("%s/%s", params.ProjectName, params.RepositoryName),
```

Comment:
So many methods depend on this utility method `GetByReference`, we need to check if there is something not well designed.